request-attrs.go

```diff
@@ -58,6 +54,11 @@ func (a FileStat) FileMode() os.FileMode {
 // Attributes parses file attributes byte blob and return them in a
 // FileStat object.
 func (r *Request) Attributes() *FileStat {
-	fs, _ := getFileStat(r.Flags, r.Attrs)
-	return fs
+	var attrs sshfx.Attributes
+
+	_ = attrs.XXX_UnmarshalByFlags(r.Flags, sshfx.NewBuffer(r.Attrs))
```

Comment:
This is was I was talking about above in the `filexfer/attrs.go` file. We need to take these two parameters separately, because really don’t want to rend out the `r.Flags` into another buffer and appending the `Attrs` slice to the end.